java/org/apache/catalina/webresources/AbstractSingleArchiveResourceSet.java

```diff
@@ -60,29 +70,41 @@ public AbstractSingleArchiveResourceSet(WebResourceRoot root, String webAppMount
         }
     }
 
+    @Override
+    protected boolean isCacheArchiveEntries() {
```

Comment:
I think if we want to submit this upstream, it might be worth trying to push the logic of reading the system property into the base class.
Right now everything extends `AbstractSingleArchiveResourceSet.java`  except for `JarWarResourceSet` which would then be the only type falling back to the default false value.

Totally fine for now though

Reply:
FWIW in an early diagnostic test I traced out the `JarWarResourceSet`, and we don't seem to interact with those at runtime, which is why I was keeping a narrower scope.  This is the only spot we've ever seen in blocked threads to date.